### PR TITLE
Spacefinder: wait for interactives to load

### PR DESCRIPTION
**Pull request by regiskuckaertz**
When an article contains interactives, spacefinder may get geometry wrong if the interactives haven't been loaded yet and therefore think there is space enough for inline slots:

picture 2

Alas, there is no way of knowing if an interactive has been "enhanced" unless we wait for the load event of the containing iframe... which may not even exist when spacefinder is running. Event delegation is a no go, so I had to use a `MutationObserver`, which was kind of fun.

picture 1

@uplne

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/article/spacefinder.js [hunk 1]
```diff
@@ -79,12 +79,54 @@ define([
...
+        var notLoaded = qwery('.element-interactive', body).filter(function (interactive) {
+            var iframe = qwery(interactive.children).filter(isIframe);
+            return !(iframe.length && isIframeLoaded(iframe[0]));
+        });
+
+        return notLoaded.length === 0 ?
+            true :
+            Promise.all(notLoaded.map(function (interactive) {
+                return new Promise(function (resolve) {
+                    new MutationObserver(function (records, instance) {
```
Comment (gtrufitt) [inline static/src/javascripts/projects/common/modules/article/spacefinder.js]: Do we polyfill this? Do we need to for IE10,9?

Comment (regiskuckaertz) [inline static/src/javascripts/projects/common/modules/article/spacefinder.js]: I'll count this as a progressive enhancement, in which case I should probably add a test for `MutationObserver`. Thanks for the shout out!

Comment (uplne) [inline static/src/javascripts/projects/common/modules/article/spacefinder.js]: progressive enhancement ftw. after that you can 🚢 👍

Comment (gtrufitt) [inline static/src/javascripts/projects/common/modules/article/spacefinder.js]: 💛 👍

**Pull request merged**